### PR TITLE
Cleanup time

### PR DESCRIPTION
**Pull request by ryanpbrewster**
@andrewconner @leogrim @camhashemi

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/model/Library.scala [hunk 1]
```diff
@@ -286,30 +291,27 @@ object LibrarySlug {
...
+    Writes { o => JsString(o.value) }
+  )
 }
 
 sealed abstract class LibraryKind(val value: String, val priority: Int) {
   def compare(other: LibraryKind): Int = this.priority compare other.priority
 }
 
-object LibraryKind {
+object LibraryKind extends Enumerator[LibraryKind] {
```
Comment (camhashemi) [inline kifi-backend/modules/common/app/com/keepit/model/Library.scala:314]: this is cool

File: kifi-backend/modules/common/app/com/keepit/model/Library.scala [hunk 2]
```diff
@@ -386,33 +381,27 @@ case class BasicLibraryDetails(
...
+  def get(str: String) = all.find(c => c.value == str || c.hex == str)
+  def apply(str: String): LibraryColor = get(str).getOrElse(throw new Exception(s"Unknown library color $str"))
+
+  implicit val format: Format[LibraryColor] = Format(
+    EnumFormat.reads(get, all.map(_.value) ++ all.map(_.hex)),
+    Writes { o => JsString(o.hex) }
+  )
 
-  val AllColors: Seq[LibraryColor] = Seq(BLUE, SKY_BLUE, GREEN, ORANGE, RED, MAGENTA, PURPLE)
+  val AllColors: Seq[LibraryColor] = all.toSeq.sortBy(_.value)
```
Comment (camhashemi) [inline kifi-backend/modules/common/app/com/keepit/model/Library.scala:404]: no `Set`?

Comment (ryanpbrewster) [inline kifi-backend/modules/common/app/com/keepit/model/Library.scala:404]: It was a `Seq` before. This actually might break if something relies on the fact that it's ordered in a specific way.

Comment (ryanpbrewster) [inline kifi-backend/modules/common/app/com/keepit/model/Library.scala:404]: Just checked, it's not used anywhere. :shrug:

Comment (camhashemi) [inline kifi-backend/modules/common/app/com/keepit/model/Library.scala:404]: I don't care too much, as long as the old ordering is the same as the new one?

Comment (camhashemi) [inline kifi-backend/modules/common/app/com/keepit/model/Library.scala:404]: you never know when you're going to need `SKY_BLUE` or `MAGENTA` handy

**Comment by camhashemi** [comment]
this is great, so happy

**Pull request merged**